wgpu/src/util/belt.rs

```diff
@@ -51,11 +57,55 @@ impl StagingBelt {
     /// * 1-4 times less than the total amount of data uploaded per submission
     ///   (per [`StagingBelt::finish()`]); and
     /// * bigger is better, within these bounds.
+    ///
+    /// The buffers returned by this staging belt will have the buffer usages
+    /// [`MAP_READ | MAP_WRITE`](BufferUsages).
     pub fn new(device: Device, chunk_size: BufferAddress) -> Self {
+        Self::new_with_buffer_usages(device, chunk_size, BufferUsages::COPY_SRC)
+    }
+
+    /// Create a new staging belt.
+    ///
+    /// The `chunk_size` is the unit of internal buffer allocation; writes will be
+    /// sub-allocated within each chunk. Therefore, for optimal use of memory, the
+    /// chunk size should be:
+    ///
+    /// * larger than the largest single [`StagingBelt::write_buffer()`] operation;
+    /// * 1-4 times less than the total amount of data uploaded per submission
+    ///   (per [`StagingBelt::finish()`]); and
+    /// * bigger is better, within these bounds.
+    ///
+    /// `buffer_usages` specifies with which [`BufferUsages`] the staging buffers
+    /// will be created. [`MAP_WRITE`](BufferUsages::MAP_WRITE) will be added
+    /// automatically. The method will panic if the combination of usages is not
+    /// supported. Because [`MAP_WRITE`](BufferUsages::MAP_WRITE) is implied, only
+    /// [`COPY_SRC`](BufferUsages::COPY_SRC) can be used, except if
+    /// [`Features::MAPPABLE_PRIMARY_BUFFERS`] is enabled.
```

Comment:
```suggestion
    /// supported. Because [`MAP_WRITE`](BufferUsages::MAP_WRITE) is implied, the allowed usages
    /// depends on if [`Features::MAPPABLE_PRIMARY_BUFFERS`] is enabled.
    /// - If enabled: any usage is valid.
    /// - If disabled: only [`COPY_SRC`](BufferUsages::COPY_SRC) can be used.
```
How does that sound, I think this is a bit clearer.

Reply:
Better :)

I strictly any usage allowed with `MAPPABLE_PRIMARY_BUFFERS`? And what if that would change in the future? Maybe we should just link to the documentation of that feature.

Reply:
Any anything is allowed with `MAPPABLE_PRIMARY_BUFFERS` - that shouldn't change in the future. This was one of the first features we had and has remained largely unchanged since.